x-pack/plugin/otel-data/src/main/resources/component-templates/otel@mappings.yaml

```diff
@@ -0,0 +1,67 @@
+---
+version: ${xpack.oteldata.template.version}
+_meta:
+  description: Default mappings for all OTel data streams
+  managed: true
+template:
+  mappings:
+    date_detection: false
+    dynamic: strict
+    properties:
+      "@timestamp":
+        type: date_nanos
+      data_stream.type:
+        type: constant_keyword
+      data_stream.dataset:
+        type: constant_keyword
+      data_stream.namespace:
+        type: constant_keyword
+      attributes:
+        type: passthrough
+        dynamic: true
+        priority: 10
+        time_series_dimension: true
+      dropped_attributes_count:
+        type: long
+      scope:
+        properties:
+          name:
+            type: keyword
+            ignore_above: 1024
+# TODO:
+#Causes error during YAML test
+# [2024-07-19T04:36:51,833][ERROR][o.e.x.c.t.IndexTemplateRegistry] [test-cluster-0] error adding index template [otel@mappings] for [otel] org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: No handler for type [version] declared on field [version]
```

Comment:
You'll need to add a test dependency on the `mapper-version` module. See also

https://github.com/elastic/elasticsearch/blob/ba3501ae29b010643577379446bc49fdf4775320/x-pack/plugin/profiling/build.gradle#L23